packages/react-server-cli/src/startServer.js

```diff
@@ -160,40 +195,72 @@ const startHotLoadJsServer = (compiler, port, httpsOptions) => {
 		cert: httpsOptions ? httpsOptions.cert : undefined,
 		ca: httpsOptions ? httpsOptions.ca : undefined,
 	});
-	const serverStartedPromise = new Promise((resolve) => {
-		jsServer.listen(port, () => resolve() );
+	const serverStartedPromise = new Promise((resolve, reject) => {
+		jsServer.listen(port, (e) => {
+			if (e) {
+				reject(e);
+				return;
+			}
+			resolve();
+		});
 	});
-	return Promise.all([compiledPromise, serverStartedPromise])
-		.then(() => logger.info(`Started hot reload JavaScript server over ${httpsOptions ? "HTTPS" : "HTTP"} on port ${port}`));
+	return {
+		stop: serverToStopPromise(jsServer),
+		started: Promise.all([compiledPromise, serverStartedPromise])
+			.then(() => logger.info(`Started hot reload JavaScript server over ${httpsOptions ? "HTTPS" : "HTTP"} on port ${port}`)),
+	};
 };
 
+// takes in the err and stats object returned by a webpack compilation and returns
+// an error object if something serious happened, or null if things are ok.
 const handleCompilationErrors = (err, stats) => {
 	if(err) {
 		logger.error("Error during webpack build.");
 		logger.error(err);
-		throw new Error(err);
+		return new Error(err);
```

Comment:
Helpfully, it's completely non-standard! From [the MDN doc on Error.prototype.stack](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Stack):

> Different browsers set this value at different times. For example, Firefox sets it when creating an Error object, while PhantomJS sets it only when throwing the Error, and MSDN docs also seem to match the PhantomJS implementation.

Given that the method that calls this method doesn't throw the error (because it uses it for a Promise `reject()`); I think the question is moot? (Although admittedly, promise error handling consistently makes me feel stupid, so I may be thinking about this wrong.)